src/parse.rs

```diff
@@ -0,0 +1,407 @@
+use crate::{
+    errors::{Error, Result},
+    RSAPrivateKey, RSAPublicKey,
+};
+use simple_asn1::{ASN1Block, ASN1DecodeErr, BigUint, OID};
+
+use std::convert::TryFrom;
+
+impl From<ASN1DecodeErr> for Error {
+    fn from(e: ASN1DecodeErr) -> Error {
+        Error::ParseError {
+            reason: format!("{}", e),
+        }
+    }
+}
+
+#[cfg(feature = "pem")]
+impl TryFrom<pem::Pem> for RSAPrivateKey {
+    type Error = Error;
+
+    /// Parses a `PKCS8` or `PKCS1` encoded RSA Private Key.
+    ///
+    /// Expects one of the following `pem` headers:
+    /// - `-----BEGIN PRIVATE KEY-----`
+    /// - `-----BEGIN RSA PRIVATE KEY-----`
+    fn try_from(pem: pem::Pem) -> Result<RSAPrivateKey> {
+        match &*pem.tag {
+            "RSA PRIVATE KEY" => parse_private_key_pkcs1(&pem.contents),
+            "PRIVATE KEY" => parse_private_key_pkcs8(&pem.contents),
+            _ => Err(Error::ParseError {
+                reason: format!("unexpected tag: {}", pem.tag),
+            }),
+        }
+    }
+}
+
+#[cfg(feature = "pem")]
+impl TryFrom<pem::Pem> for RSAPublicKey {
+    type Error = Error;
+
+    /// Parses a `PKCS8` or `PKCS1` encoded RSA Public Key.
+    ///
+    /// Expects one of the following `pem` headers:
+    /// - `-----BEGIN PUBLIC KEY-----`
+    /// - `-----BEGIN RSA PUBLIC KEY-----`
+    fn try_from(pem: pem::Pem) -> Result<RSAPublicKey> {
```

Comment:
same here

Reply:
@dignifiedquire I've added examples for all of the new functions.